iree/compiler/Dialect/HAL/Target/ROCM/ROCMTarget.cpp

```diff
@@ -149,14 +136,20 @@ class ROCMTargetBackend final : public TargetBackend {
         workgroup_size = {1, 1, 1};
       }
       workgroupSizes.push_back(workgroup_size);
+      // For GPU kernels,
+      // 1. Insert AMDGPU_KERNEL calling convention.
+      // 2. Insert amdgpu-flat-workgroup-size(1, 256) attribute.
+      // 3. Insert amdgpu-implicitarg-num-bytes=56 (which must be set on OpenCL
+      // and HIP kernels per Clang)
       llvmFunc->setCallingConv(llvm::CallingConv::AMDGPU_KERNEL);
-      llvmFunc->addFnAttr("amdgpu-flat-work-group-size", "1, 1024");
+      llvmFunc->addFnAttr("amdgpu-flat-work-group-size", "1, 256");
+      llvmFunc->addFnAttr("amdgpu-implicitarg-num-bytes", "56");
```

Comment:
Do we need this? I don't think we currently use any of the implicit argument? I'm not sure if this has a perf impact. Are you just trying to be conservative?

Reply:
Yeah, should we remove it then or is it fine to just keep it?

Reply:
yeah I’d remove it as it may have a performance impact

Reply:
sounds good